File: tests/fixtures/joern_raw_v2/java/Flow.java
package rawprobe;

final class Flow {
    private String saved;

    static String helper(String left, String right) {
        String combined = left + ":" + right;
        return combined;
    }

    static String helper(int left, String right) {
        return right + left;
    }

    String instance(String value) {
        saved = value;
        return helper(value, saved);
    }

    static String variadic(String first, String... rest) {
        return first + rest[0];
    }

    String caller(String input, String[] values) {
        String named = helper(input, values[0]);
        String changed = helper(1, named);
        String result = this.instance(changed);
        return variadic(result, input, named);
    }

    String external(java.util.function.Function<String, String> target, String value) {
        return target.apply(value);
    }
}

File: tests/fixtures/joern_raw_v2/python/flow.py
"""Static-parser input only. This file is never imported or executed."""


def helper(left, right="suffix"):
    combined = left + ":" + right
    return combined


def variadic(first, *rest, **named):
    return first + rest[0] + named["suffix"]


class Flow:
    def __init__(self, cursor):
        self.cursor = cursor
        self.saved = ""

    def instance(self, value):
        self.saved = value
        return helper(value, right=self.saved)

    def caller(self, user_input, values, named):
        first = helper(user_input, values[0])
        second = helper(right=first, left=user_input)
        third = helper(second)
        self.cursor.execute(self.instance(third))
        return self.cursor.fetchall()

    def splat(self, values, named):
        return variadic("prefix", *values, **named)


def unknown(target, value):
    return target(value)
